### PR TITLE
build: add git branch name in tikv version and starting log

### PR DESCRIPTION
**Pull request by overvenus**
```
$ target/debug/tikv-server -V
TiKV 0.0.1
Git Commit Hash:   2fcd5b624b43fd23a70fb831ce809d2e7f86832b
Git Commit Branch: ov/branch-info
UTC Build Time:    2017-08-14 10:29:42
Rust Version:      1.21.0-nightly (215e0b10e 2017-08-08)
```

CC pingcap/tidb#4168

### CODE REVIEW

**Comment by siddontang** [comment]
PTAL  @BusyJay @zhangjinpeng1987

File: build.rs [hunk 1]
```diff
@@ -48,6 +50,12 @@ fn commit_hash() -> String {
     cmd.output().ok().and_then(|o| String::from_utf8(o.stdout).ok()).unwrap_or("None".to_owned())
 }
 
+fn branch_info() -> String {
+    let mut cmd = Command::new("git");
```
Comment (zhangjinpeng1987) [inline build.rs]: Is this means the target machines should have git installed and the internet works.

**Comment by disksing** [comment]
LGTM.

File: build.rs (hunk 1 above)
Comment (disksing) [inline build.rs]: No. `build.rs` is for compile.

**Comment by BusyJay** [comment]
What does the branch name here mean? For example, master contains commits a, b, c, branch example contains a, b, d, e. Then for commit d, what's the branch name of if? How about b or a?

**Comment by siddontang** [comment]
@BusyJay 

We just follow the rule of TiDB. We only need to care whether the binary belongs to which branch, master, RC or GA. Then we will seek the commit in the branch.

**Comment by BusyJay** [comment]
LGTM

**Comment by siddontang** [comment]
Please fix the conflict @overvenus

**Comment by siddontang** [comment]
@overvenus please cherry pick to RC4

**Pull request merged**